cfn-resources/private-endpoint-service/cmd/resource/resource.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	resource_constats "github.com/mongodb/mongodbatlas-cloudformation-resources/private-endpoint-service/cmd/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/private-endpoint-service/cmd/resource/steps/privateendpointservice"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	providerName = "AWS"
+)
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-private-endpoint")
+}
+
+var CreateRequiredFields = []string{constants.ProjectID, constants.Region}
+var ReadRequiredFields = []string{constants.ProjectID, constants.ID, constants.Region}
+var UpdateRequiredFields []string
+var DeleteRequiredFields = []string{constants.ProjectID, constants.ID}
+var ListRequiredFields = []string{constants.ProjectID}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	if errEvent := validator.ValidateModel(CreateRequiredFields, currentModel); errEvent != nil {
+		_, _ = logger.Warnf("Validation Error")
+		return *errEvent, nil
+	}
+
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+	mongodbClient, pe := util.NewMongoDBClient(req, currentModel.Profile)
+	if pe != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
+		return *pe, nil
+	}
+
+	status, pe := getProcessStatus(req)
+	if pe != nil {
+		return *pe, nil
+	}
+
+	switch status {
+	case resource_constats.Init:
+		pe := privateendpointservice.Create(*mongodbClient, *currentModel.Region, *currentModel.ProjectId)
```

Comment:
It there a reason we limit the cloud provider to AWS in private endpoint service, but leave it configurable in private endpoint?

Reply:
so in mi mind the user might need a service for other provider,
to create a complete private endpoint you need:
1- create the service for an specific provider (private-endpoint-service resource)
2-create the aws private endpoint (this the user must use the aws private endpoint cfn resource)
3-create an atlas private endpoint

leaving the CloudProvider we can leave a door so the user can at least create the step 1 for the other prividers, and then setup tha azure or any other connection by hand

Reply:
From what you mention in point 1, we would leave the provider configurable in private-endpoint-service resource. However, as of the current implementation private-endpoint-service is fixed to AWS, while private-endpoint has a configurable provider property so I'm confused.

Reply:
yes you are right on this one, they are switch... ill change that